File: frontend-Application/src/services/admin/batchService.js
import adminAxios from "../axios/adminAxios";

export const getAllBatches = async () => {
    return await adminAxios.get("/admin/batches");
};

export const getBatchById = async (id) => {
    return await adminAxios.get(`/admin/batches/${id}`);
};

export const createBatch = async (batch) => {
    return await adminAxios.post("/admin/batches", batch);
};

export const updateBatch = async (id, batch) => {
    return await adminAxios.put(`/admin/batches/${id}`, batch);
};

export const deleteBatch = async (id) => {
    return await adminAxios.delete(`/admin/batches/${id}`);
};

File: oep_spring_backend/src/main/java/com/oep/service/BatchService.java
package com.oep.service;

import java.util.List;
import com.oep.entities.Batch;

public interface BatchService {
    List<Batch> getAllBatches();

    Batch getBatchById(Long id);

    Batch createBatch(Batch batch);

    Batch updateBatch(Long id, Batch batch);

    void deleteBatch(Long id);
}
